File: work/DSA/c++/recursion_1/lastIndexOfANumber.cpp
#include <bits/stdc++.h>
using namespace std;

int lastIndex(int input[], int n, int x)
{
    if (n == 0)
    {
        return -1;
    }
    int ans = lastIndex(input + 1, n - 1, x);
    if (ans != -1)
    {
        return ans + 1;
    }
    else if (input[0] == x)
    {
        return 0;
    }
    else
    {
        return -1;
    }
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;

    int *input = new int[n];

    for (int i = 0; i < n; i++)
    {
        cin >> input[i];
    }

    int x;

    cin >> x;

    cout << lastIndex(input, n, x) << endl;
}

File: work/DSA/javascript/recursion_1/lastIndexOfANumber.js
let n = 4
let input = [9, 8, 10, 8]
let x = 8

function lastIndex(input, n, x) {
    if (n === 0) {
        return -1;
    }
    let subAnswer = lastIndex(input.slice(1), n - 1, x)
    if (subAnswer !== -1) {
        return subAnswer + 1;
    }
    else if (input[0] === x) {
        return 0;
    }
    else {
        return -1;
    }

}

let answer = lastIndex(input, n, x)
console.log(answer)